### ISSUE TITLE
Add `start_lr` option to `LRFinder` 

### ISSUE THREAD

**Issue by KickItLikeShika** [pytorch/ignite#2105]
## 🚀 Feature

Currently we choose the start learning rate for the Learning Rate Finder as the learning rate specified with the optimizer
in case of linear `step_mode`
https://github.com/pytorch/ignite/blob/master/ignite/handlers/lr_finder.py#L124
And also in case of exp `step_mode` (we use torch `_LRScheduler`) they specify their `base_lrs` as the given learning rates given with the optimizer also 
https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/torch/optim/lr_scheduler.py#L35
Which is not the best choice if the learning rate with the optimizer is not a small value.

The idea is to add `start_lr` argument to the `attach` method in the `LRFinder` to be used in both modes (linear, exp), that would be a nice improvement to to get a better results from the LRFinder.
